### PR TITLE
Use HA proxy in smart update

### PR DESCRIPTION
**Pull request by blez**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| big-data | passed |
| recreate | passed |
| storage | passed |
| scaling | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | failed |
| limits | passed |
| upgrade | passed |
| demand-backup | failed |
| self-healing-advanced | passed |
| monitoring | passed |
| scheduled-backup | failed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| security-context | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| upgrade | passed |
| recreate | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| restore-to-encrypted-cluster | failed |
| security-context | passed |
| init-deploy | passed |
| big-data | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| demand-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | failed |
| init-deploy | passed |
| storage | passed |
| limits | passed |
| self-healing | passed |
| monitoring | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| big-data | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | passed |

File: pkg/controller/pxc/upgrade.go [hunk 1]
```diff
@@ -225,18 +227,23 @@ func (r *ReconcilePerconaXtraDBCluster) applyNWait(cr *api.PerconaXtraDBCluster,
...
+	}
+
 	return nil
 }
 
 func (r *ReconcilePerconaXtraDBCluster) waitUntilOnline(cr *api.PerconaXtraDBCluster, sfsName string, pod *corev1.Pod, waitLimit int) error {
+	if cr.Spec.HAProxy.Enabled {
+		time.Sleep(5 * time.Second)
+		return nil
+	}
```
Comment (dAdAbird) [inline pkg/controller/pxc/upgrade.go:245]: It would be great to have a comment describing why we need just to wait for 5 sec having HAProxy

**Pull request merged**